os/arch/arm/src/amebad/Kconfig

```diff
@@ -12,493 +12,50 @@ choice
 
 config ARCH_CHIP_RTL8721CSM
 	bool "RTL8721CSM"
-	select RTL8721D
 	select ARCH_CORTEXM33
-	select CHIP_A_CUT
-	select KERNEL
-	select SOC_PS_EN
-	select SDIO_DEVICE_EN
-	select MBED_API_EN
-	select WIFI_EN
-	select NETWORK
-	select USE_MBEDTLS_ROM
-	select MBED_TLS_ENABLED
-	select DEBUG_LOG
-	select TOOLCHAIN_ASDK
-	select LINK_ROM_SYMB
+	select AMEBAD_WIFI
+	select ARMV8M_HAVE_ICACHE
+	select ARMV8M_HAVE_DCACHE
 
 endchoice
 
-choice
-	prompt "Product"
-	default RTL8721D
-
-config RTL8721D
```

Comment:
Could you let me know the reason why you remove those config?
There are many codes to check `CONFIG_RTL8721D` conditional in your source codes.
```
Autoconf.h (os\board\rtl8721csm\src\component\common\drivers\wlan\realtek\include):		#ifndef CONFIG_RTL8721D 
Autoconf.h (os\board\rtl8721csm\src\component\common\drivers\wlan\realtek\include):			#define CONFIG_RTL8721D
Autoconf.h (os\drivers\wireless\realtek\rtk\include):#ifndef CONFIG_RTL8721D
Autoconf.h (os\drivers\wireless\realtek\rtk\include):#define CONFIG_RTL8721D
Defconfig (build\configs\rtl8721csm\hello):CONFIG_RTL8721D=y
Defconfig (build\configs\rtl8721csm\tc):CONFIG_RTL8721D=y
Drv_types.h (os\drivers\wireless\realtek\rtk\include):#if defined(CONFIG_RTL8821C) || defined(CONFIG_RTL8195B) || defined(CONFIG_RTL8721D) || defined(CONFIG_RTL8710C)
Drv_types.h (os\drivers\wireless\realtek\rtk\include):#if defined(CONFIG_RTL8195B) || defined(CONFIG_RTL8721D) || defined(CONFIG_RTL8710C)
Hal_btcoex_wifionly.h (os\drivers\wireless\realtek\rtk\include):#if defined(CONFIG_RTL8721D) || defined(CONFIG_RTL8710C) || defined(CONFIG_RTL8821C) || defined(CONFIG_RTL8723B) || defined(CONFIG_RTL8703B)
Hal_com.h (os\drivers\wireless\realtek\rtk\include):#if defined(CONFIG_RTL8188F) || defined(CONFIG_RTL8192E) || defined(CONFIG_RTL8703B) || defined(CONFIG_RTL8723B) || defined(CONFIG_RTL8723D) || defined(CONFIG_RTL8711B) || defined(CONFIG_RTL8821C) || defined(CONFIG_RTL8721D) || defined(CONFIG_RTL8195B) || defined(CONFIG_RTL8710C) || defined(CONFIG_RTL8192F)
Hal_data.h (os\drivers\wireless\realtek\rtk\src\hal):#ifdef CONFIG_RTL8721D
Hal_data.h (os\drivers\wireless\realtek\rtk\src\hal):#if defined(CONFIG_RTL8821C) || defined(CONFIG_RTL8195B) || defined(CONFIG_RTL8721D) || defined(CONFIG_RTL8710C)
Hal_data.h (os\drivers\wireless\realtek\rtk\src\hal):#if defined(CONFIG_RTL8821C) || defined(CONFIG_RTL8192F) || defined(CONFIG_RTL8192E) || defined(CONFIG_RTL8195B) || defined(CONFIG_RTL8721D) || defined(CONFIG_RTL8710C)
Hal_data.h (os\drivers\wireless\realtek\rtk\src\hal):#if defined(CONFIG_RTL8821C) || defined(CONFIG_RTL8195B) || defined(CONFIG_RTL8721D) || defined(CONFIG_RTL8710C)
Hal_data.h (os\drivers\wireless\realtek\rtk\src\hal):#if (defined(CONFIG_RTL8711B) || defined(CONFIG_RTL8188F) || defined(CONFIG_RTL8192E) || defined(CONFIG_RTL8723B) || defined(CONFIG_RTL8703B) || defined(CONFIG_RTL8723D) || defined(CONFIG_RTL8821C)) || defined(CONFIG_RTL8721D) || defined(CONFIG_RTL8195B) || defined(CONFIG_RTL8188E) || defined(CONFIG_RTL8710C) || defined(CONFIG_RTL8192F)
Hal_data.h (os\drivers\wireless\realtek\rtk\src\hal):#if defined(CONFIG_RTL8195B) || defined(CONFIG_RTL8721D) || defined(CONFIG_RTL8710C)
Hal_intf.h (os\drivers\wireless\realtek\rtk\include):#elif defined(CONFIG_RTL8721D)
Lxbus_ops.h (os\drivers\wireless\realtek\rtk\include):#elif defined(CONFIG_RTL8721D)
Platform_autoconf.h (os\board\rtl8721csm\src\project\realtek_amebad_va0_example\inc\inc_hp):#define CONFIG_RTL8721D 1
Rtw_btcoex_wifionly.h (os\drivers\wireless\realtek\rtk\include):#if defined(CONFIG_RTL8721D) || defined(CONFIG_RTL8710C) || defined(CONFIG_RTL8821C)
Rtw_eeprom.h (os\drivers\wireless\realtek\rtk\include):#if defined(CONFIG_RTL8821C) || defined(CONFIG_RTL8192F) || defined(CONFIG_RTL8721D)
Rtw_xmit.h (os\drivers\wireless\realtek\rtk\include):#if defined(CONFIG_RTL8812A) || defined(CONFIG_RTL8821A) || defined(CONFIG_RTL8723B) || defined(CONFIG_RTL8703B) || defined(CONFIG_RTL8195A) || defined(CONFIG_RTL8711B) || defined(CONFIG_RTL8188F) || defined(CONFIG_RTL8192E) || defined(CONFIG_RTL8721D) || defined(CONFIG_RTL8710C) || defined(CONFIG_RTL8723D) || defined(CONFIG_RTL8192F)
Rtw_xmit.h (os\drivers\wireless\realtek\rtk\include):#if defined(CONFIG_RTL8195A) || defined(CONFIG_RTL8711B) || defined(CONFIG_RTL8721D) || defined(CONFIG_RTL8195B) || defined(CONFIG_RTL8710C) // buffer descriptor architecture
Wifi_interactive_mode.c (os\board\rtl8721csm\src\component\common\api):#if defined(CONFIG_RTL8195A) || defined(CONFIG_RTL8711B) || defined(CONFIG_RTL8721D)
Wifi_interactive_mode.c (os\drivers\wireless\realtek\wifi_util_interface):#if defined(CONFIG_RTL8195A) || defined(CONFIG_RTL8711B) || defined(CONFIG_RTL8721D)
Wifi_interactive_mode.c (os\drivers\wireless\realtek\wifi_util_interface):#if defined(CONFIG_RTL8195A) || defined(CONFIG_RTL8711B) || defined(CONFIG_RTL8721D)
```

Reply:
As you know, CFLAGS has character length limitation. If it could be supported by config, it is better to use config instead of -D in Makefile.

Reply:
@sunghan-chang The CONFIG_RTL8721D is defined in platform_autoconf.h, we do not want end user to control this setting.

Reply:
ok